native/src/main/java/io/ballerina/stdlib/oracledb/utils/ConverterUtils.java

```diff
@@ -139,6 +147,89 @@ public static Array convertVarray(Connection connection, Object value)
+                        ValueCreator.createArrayValue(io.ballerina.stdlib.sql.utils.Utils.FLOAT_ARRAY);
+                for (int i = 0; i < dataArray.length; ++i) {
+                    typedArray.add(i, dataArray[i] != null ? ((BigDecimal) dataArray[i]).doubleValue() : null);
+                }
+                return typedArray;
+            case Constants.Types.BallerinaArrayTypes.OPTIONAL_DECIMAL:
+            case io.ballerina.stdlib.sql.Constants.ArrayTypes.DECIMAL:
+            case Constants.Types.BallerinaArrayTypes.ANYDATA:
+                typedArray = containsNull ? parameterProcessor.
+                        createEmptyBBRefValueArray(PredefinedTypes.TYPE_DECIMAL) :
+                        ValueCreator.createArrayValue(io.ballerina.stdlib.sql.utils.Utils.DECIMAL_ARRAY);
+                for (int i = 0; i < dataArray.length; ++i) {
+                    typedArray.add(i, dataArray[i] != null ?
+                            ValueCreator.createDecimalValue((BigDecimal) dataArray[i]) : null);
+                }
+                return typedArray;
+            case Constants.Types.BallerinaArrayTypes.OPTIONAL_BOOLEAN:
+            case io.ballerina.stdlib.sql.Constants.ArrayTypes.BOOLEAN:
+                typedArray = containsNull ? parameterProcessor.
+                        createEmptyBBRefValueArray(PredefinedTypes.TYPE_BOOLEAN) :
+                        ValueCreator.createArrayValue(io.ballerina.stdlib.sql.utils.Utils.BOOLEAN_ARRAY);
+                for (int i = 0; i < dataArray.length; ++i) {
+                    typedArray.add(i, dataArray[i] != null ? dataArray[i].equals(BigDecimal.ONE) : null);
+                }
+                return typedArray;
+            default:
+                throw new DataError(String.format("Cannot cast varray to type: %s", type));
+        }
+    }
+
+    public static BArray convertStringArrayToBallerinaType(DefaultResultParameterProcessor parameterProcessor,
+                                                           Object[] dataArray, Type type, boolean containsNull)
+            throws DataError {
+        if (type.toString().equals(Constants.Types.BallerinaArrayTypes.OPTIONAL_STRING) ||
+                type.toString().equals(io.ballerina.stdlib.sql.Constants.ArrayTypes.STRING) ||
+                type.toString().equals(Constants.Types.BallerinaArrayTypes.ANYDATA)) {
+            BArray stringDataArray = containsNull ? parameterProcessor.
+                    createEmptyBBRefValueArray(PredefinedTypes.TYPE_STRING) :
+                    ValueCreator.createArrayValue(io.ballerina.stdlib.sql.utils.Utils.STRING_ARRAY);
+            for (int i = 0; i < dataArray.length; i++) {
+                stringDataArray.append(dataArray[i] != null ? fromString(dataArray[i].toString()) : null);
+            }
+            return stringDataArray;
+        } else {
+            throw new DataError(String.format("Cannot cast varray to type: %s", type));
```

Comment:
Same

Reply:
changed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/28/commits/b0eae79974b603113b47b8e0c88ae9dc9fb152b1